lib/Sema/CSApply.cpp

```diff
@@ -6453,6 +6453,12 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
       }
     }
 
+    // Allows for same-type coercion with existential metatypes.
```

Comment:
It’s a bit odd that coerceToType is called like this. Is toMeta an existential metatype or protocol metatype?

Reply:
Ah, I think you're right that toMeta is a protocol metatype here, so I might need to change the way I handle this.